src/wr_control_drive_science/src/TeleopScienceControl.cpp

```diff
@@ -0,0 +1,154 @@
+#include "ros/assert.h"
+#include "ros/ros.h"
+#include "std_msgs/Float32.h"
+#include "std_msgs/Bool.h"
+#include "std_msgs/UInt32.h"
+#include "std_msgs/Float64.h"
+#include "std_msgs/Int16.h"
+#include "ros/timer.h"
+#include <fcntl.h>
+#include <termios.h>
+#include <string>
+#include <array>
+
+constexpr std::uint32_t MESSAGE_CACHE_SIZE = 10;
+constexpr float TIMER_CALLBACK_DURATION = 5;
+
+ros::Publisher moistureSensor1;
+ros::Publisher moistureSensor2;
+ros::Publisher moistureSensor3;
+int baudRate;
+std::string file;
+int fd;
+
+void moistureCallback(const ros::TimerEvent &timerEvent) {
+    std::array<float, 3> buf{0, 0, 0};
+    /* Flush anything already in the serial buffer */
+    tcflush(fd, TCIFLUSH);
+    /* read up to 128 bytes from the fd */
+    int readStatus = read(fd, buf.data(), 3*sizeof(float));
+    //NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay, cppcoreguidelines-pro-type-vararg, hicpp-no-array-decay, hicpp-no-assembler, hicpp-vararg)
+    ROS_ASSERT_MSG(readStatus != -1, "Read failed");
+    std_msgs::Float32 temp;
+    temp.data = buf[0];
+    moistureSensor1.publish(temp);
+    temp.data = buf[1];
+    moistureSensor1.publish(temp);
+    temp.data = buf[2];
+    moistureSensor1.publish(temp);
```

Comment:
This should be a range-based `for` loop over whatever is in the buffer, based on the layout.  There may also be a data malformation issue by publishing all 3 readings over the same topic.